keanu-project/src/main/java/io/improbable/keanu/model/SamplingModelFitter.java

```diff
@@ -0,0 +1,41 @@
+package io.improbable.keanu.model;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.PosteriorSamplingAlgorithm;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+
+public class SamplingModelFitter<INPUT, OUTPUT> implements ModelFitter<INPUT, OUTPUT> {
```

Comment:
May be worth giving a short description of how this works and maybe how it is different to the ML/MAP estimation fitters?

Reply:
Javadoc is great - my only concern now is that because this will often be used via the `RegressionModelBuilder.withSampling()` method that we should maybe put some documentation on that too?

Reply:
good call - done.